Packs/JoeSecurity/Integrations/JoeSecurityV2/JoeSecurityV2_test.py

```diff
@@ -0,0 +1,229 @@
+    """
+    from JoeSecurityV2 import list_analysis_command
+
+    result = util_load_json('test_data/analysis_info_list.json')
+    excepted = util_load_json('test_data/analysis.json')
+
+    client = mock_client()
+    mocker.patch.object(client, 'analysis_list_paged', return_value=[])
+    mocker.patch.object(client, 'analysis_info', return_value={})
+    mocker.patch.object(client, 'analysis_info_list', return_value=result)
+
+    response = list_analysis_command(client, {})
+    for index, command_res in enumerate(response[:-1]):
+        assert command_res.indicator.dbot_score.indicator == excepted.get('DBotScore')[index].get('Indicator')
+        assert command_res.indicator.dbot_score.score == excepted.get('DBotScore')[index].get('Score')
+        if isinstance(command_res.indicator, Common.File):
+            assert command_res.indicator.sha1 == excepted.get('File')[index].get('SHA1')
+        else:
+            assert command_res.indicator.url == excepted.get('URL').get('Data')
+    assert response[-1].outputs == excepted.get('Joe').get('Analysis')
+
+
+@pytest.mark.parametrize('web_id,file_type', [('1', 'html'), ('2', 'json'), ('3', 'pcap')])
+def test_download_report_command(mocker, web_id, file_type):
+    """
+    Given:
+        - An app client object, web id and file type.
+    When:
+        - download report method has been called.
+    Then:
+        - Ensure the corresponding infoFile has been created.
+    """
+    from JoeSecurityV2 import download_report_command
+
+    client = mock_client()
+
+    mocker.patch.object(client, 'analysis_download', return_value=('test_report', 'html_test_report'))
+    response = download_report_command(client, {'webid': web_id, 'type': file_type})
+    assert response.get('File') == f'{web_id}_report.{file_type}'
+
+
+def test_download_sample_command(mocker):
+    """
+    Given:
+        - An app client object and web ids.
+    When:
+        - download report method has been called.
+    Then:
+        - Ensure the corresponding File has been created.
+    """
+    from JoeSecurityV2 import download_sample_command
+
+    client = mock_client()
+    mocker.patch.object(client, 'analysis_download', return_value=('test_sample', 'test_sample'))
+
+    for web_id in range(0, 3):
+        response = download_sample_command(client, {'webid': web_id})
+        assert response.get('File') == f'{web_id}.dontrun'
+
+
+def test_search_command(mocker):
+    """
+    Given:
+        - An app client object and fake query.
+    When:
+        - The search method has been called.
+    Then:
+        - Ensure the corresponding readable output is returned.
+    """
+    from JoeSecurityV2 import search_command
+
+    client = mock_client()
+
+    mocker.patch.object(client, 'analysis_search', return_value=[])
+    response = search_command(client, {'query': 'test.com'})[0]
+    assert response.readable_output == 'No Results were found.'
```

Comment:
I'd check there is no data in the outputs (for similar reasons to what I wrote about the exceptions)